probe-rs/src/config/memory.rs

```diff
@@ -66,7 +66,11 @@ pub trait MemoryRange {
 impl MemoryRange for Range<u32> {
     /// Returns true if `self` contains `range` fully.
     fn contains_range(&self, range: &Range<u32>) -> bool {
-        self.contains(&range.start) && self.contains(&(range.end - 1))
```

Comment:
What's the reason for this change? Is it related to the M7 or the chips?

Reply:
For some reason there would be sections that resulted range.end to be 0, in that case range.end -1 would cause an integer underflow as range.end ist a u32.

Furthermore, even if it wasn't as you're expecting files that come from some source you don't control here it would always be good to check for potential under or overflows just in case someone should produce a binary that does trigger the bug like mine do apparently.

Reply:
Good catch!

Can you also add this to the `intersects_range` method below? The same problem could happen there.

Reply:
Done!